sched/job_def.thrift

```diff
@@ -9,26 +9,26 @@
 #
 
 # To Generate files run from this (github.com/scootdev/scoot/sched) directory
-#     1. thrift --gen go:package_prefix=github.com/scootdev/scoot/sched/gen-go/,package=schedthrift,thrift_import=github.com/apache/thrift/lib/go/thrift JobDef.thrift
+#     1. thrift --gen go:package_prefix=github.com/scootdev/scoot/sched/gen-go/,package=schedthrift,thrift_import=github.com/apache/thrift/lib/go/thrift job_def.thrift
 
 
 struct Command {
-  1: list<string> argv,
-  2: map<string, string> envVars,
-  3: i64 timeout,
-  4: string snapshotId,
+  1: required list<string> argv,
+  2: optional map<string, string> envVars,
+  3: optional i64 timeout,
+  4: required string snapshotId,
 }
 
 struct TaskDefinition {
-  1: Command command,
+  1: required Command command,
 }
 
 struct JobDefinition {
-  1: string jobType,
-  2: map<string, TaskDefinition> tasks,
+  1: optional string jobType,
```

Comment:
They don't have to.  In theory you could send us a job with no tasks, it'll work fine, just nothing will be ran,  Also we don't currently use jobType, but may in the future?